packages/eas-cli/src/commandUtils/easCommand.ts

```diff
@@ -0,0 +1,37 @@
+import { Command } from '@oclif/command';
+
+import { AnalyticsEvent, flushAsync, initAsync, logEvent } from '../analytics';
+import { getUserAsync } from '../user/User';
+import { ensureLoggedInAsync } from '../user/actions';
+
+export default abstract class EasCommand extends Command {
+  /**
+   * When user data is unavailable locally, determines if the command will
+   * request it from the backend
+   * @returns boolean - if authentication is required
+   */
+  requiresAuthentication(): boolean {
+    return true;
+  }
+
+  async init() {
+    await initAsync();
+    if (this.requiresAuthentication()) {
```

Comment:
added in this check to make the class more generic - basically we eventually want all commands to subclass this for common boilerplate, so we need to change its behavior per command.  

Right now there are two cases we need to handle:
- a command requires prompting users to log in if they are not already logged in
- a command does not require authentication